File: python_solution/award_types/BlueCompare.py
class BlueCompare:
    def __init__(self, obj):
        self.obj = obj

    @property
    def current_value(self):
        if self.obj.expires_in <= 0:
            return -self.obj.quality
        else:
            if self.obj.expires_in <= 5:
                return 3
            if self.obj.expires_in <= 10:
                return 2
            else:
                return 1

File: ruby_solution/award_types/BlueCompare.rb
class BlueCompare
  attr_accessor :obj

  def initialize(obj)
    self.obj = obj
  end

  def current_value
    self.obj.expires_in <= 0 ? -self.obj.quality : (self.obj.expires_in <= 10 ? (self.obj.expires_in <= 5 ? 3 : 2) : 1)
  end
end
